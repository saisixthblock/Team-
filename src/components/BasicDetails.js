import React, { Component } from "react";
import { Card,CardBody, Col, FormGroup, Input, Label,Row } from "reactstrap";

class BasicDetails extends Component{
    render(){
        return(
            <div>
                <Row>
                    <Col md={6} >
                        <FormGroup>
                            <Label htmlFor="fullName">Fullname:</Label>
                            <Input type="text"
                                name="fullName"
                                placeholder="Enter the Fullname"
                                value={this.props.values.fullName}
                                onChange={(e)=>this.props.handleChange(e.target.name, e.target.value)}
                                onBlur={(e)=>this.props.handleBlur(e.target.name,e.target.value)}
                            />
                            {this.props.values.fullNameError ? <p style={{color:'red'}}>{this.props.values.fullNameError}</p>: null}
                        </FormGroup>
                    </Col>
                    <Col md={6} >
                        <FormGroup>
                            <Label htmlFor="phoneNumber">PhoneNumber:</Label>
                            <Input type="text"
                                name="phoneNumber"
                                placeholder="Enter the Phone Number"
                                value={this.props.values.phoneNumber}
                                onChange={(e)=>this.props.handleChange(e.target.name, e.target.value)}
                                onBlur={(e)=>this.props.handleBlur(e.target.name,e.target.value)}
                            />
                            {this.props.values.phoneNumberError ? <p style={{color:'red'}}>{this.props.values.phoneNumberError}</p> : null}
                        </FormGroup>
                    </Col>
                </Row>
                <Row>
                    <Col md={6}>
                        <FormGroup>
                            <Label htmlFor="email">Email Id:</Label>
                            <Input type="email"
                                name="email"
                                placeholder="Enter the email ID"
                                value={this.props.values.email}
                                onChange={(e)=>this.props.handleChange(e.target.name, e.target.value)}
                                onBlur={(e)=>this.props.handleBlur(e.target.name,e.target.value)}
                            />
                            {this.props.values.emailError ? <p style={{color:'red'}}>{this.props.values.emailError}</p>:null}
                        </FormGroup>
                    </Col>
                    <Col md={6} >
                        <FormGroup>
                            <Label htmlFor="password">Password:</Label>
                            <Input type="password"
                                name="password"
                                placeholder="Enter the password"
                                value={this.props.values.password}
                                onChange={(e)=>this.props.handleChange(e.target.name, e.target.value)}
                                onBlur={(e)=>this.props.handleBlur(e.target.name,e.target.value)}
                            />
                            {this.props.values.passwordError ? <p style={{color:'red'}}>{this.props.values.passwordError}</p>: null}
                        </FormGroup>
                    </Col>    
                </Row>
            </div>
        )
    }
}
export default BasicDetails