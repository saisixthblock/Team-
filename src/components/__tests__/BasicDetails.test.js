import React from "react";
import BasicDetails from "../BasicDetails";
import Enzyme from 'enzyme';
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Input } from "reactstrap";

Enzyme.configure({adapter:new Adapter() });

let sampleData={
    fullName:'abcd',
    email:'abcd@gmail.com',
    password:'abcd',
    phoneNumber:'998877',
    address:'',
    city:'hyd',
    pincode:'502032',
    state:'',
    location:'',
    gender:"",
    checkbox:'',
}

describe('Testing the Basic Details component',()=>{
    let wrapper=shallow(<BasicDetails handleChange={jest.fn()} values={sampleData} handleBlur={jest.fn()}/>);
    it('Create the snapShot of basicDetails component', async ()=>{
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
    it('Testing the FullName input field',()=>{
        expect(wrapper.find(Input).at(0).prop('value')).toEqual('abcd');// in order to check the value of the input field
    });
    it('Testing the phoneNumber input field value',()=>{
        expect(wrapper.find(Input).at(1).prop('value')).toEqual('998877'); // in order to check the value of the input field
    })
    it('Checking the onChange for fullname input feild',()=>{ // inorder to check the onChange for the input field
        wrapper.find(Input).at(0).prop('onChange')({target : {value:'chandu'}});
        wrapper.find(Input).at(1).prop('onChange')({target : {value:'98486'}});
        wrapper.find(Input).at(2).prop('onChange')({target:{value:'chandu@gmail.com'}});
        wrapper.find(Input).at(3).prop('onChange')({target:{value:'12345'}}); 
    });
    it('Checking the onBlur for the input fields',()=>{ //inorder to check the onBlur for the input field
        wrapper.find(Input).at(0).prop('onBlur')({target : {value : 'chandrakanth'}});
        wrapper.find(Input).at(1).prop('onBlur')({target : {value:'9492833880'}});
        wrapper.find(Input).at(2).prop('onBlur')({target:{value:'chandrakanth@gmail.com'}});
        wrapper.find(Input).at(3).prop('onBlur')({target:{value:'chandu'}});
    })
    it('checking the number of times the input tag is used',()=>{
        expect(wrapper.find(Input)).toHaveLength(4); // inoredr to check how many times we have used the Input tag in component
    });
    it('setProps for the input field',()=>{ // setting the props for the input field
        wrapper.setProps({
            values:{
                ...sampleData,
                fullName:'chandu',
                address:'Hyd',
                phoneNumber:'98765',
            },
        });
    });
    it('Checking the values of input field after changing the properties of field',()=>{
        expect(wrapper.find(Input).at(0).prop('value')).toEqual('chandu'); // checking the fullName input field value after setProps method
        expect(wrapper.find(Input).at(1).prop('value')).toEqual('98765'); // checking the phoneNumber input field value after setProps method
    })
});