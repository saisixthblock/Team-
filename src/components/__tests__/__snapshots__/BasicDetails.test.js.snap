// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Basic Details component Create the snapShot of basicDetails component 1`] = `
<div>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      md={6}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      <FormGroup
        tag="div"
      >
        <Label
          htmlFor="fullName"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
              "xxl",
            ]
          }
        >
          Fullname:
        </Label>
        <Input
          name="fullName"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Enter the Fullname"
          type="text"
          value="abcd"
        />
      </FormGroup>
    </Col>
    <Col
      md={6}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      <FormGroup
        tag="div"
      >
        <Label
          htmlFor="phoneNumber"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
              "xxl",
            ]
          }
        >
          PhoneNumber:
        </Label>
        <Input
          name="phoneNumber"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Enter the Phone Number"
          type="text"
          value="998877"
        />
      </FormGroup>
    </Col>
  </Row>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      md={6}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      <FormGroup
        tag="div"
      >
        <Label
          htmlFor="email"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
              "xxl",
            ]
          }
        >
          Email Id:
        </Label>
        <Input
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Enter the email ID"
          type="email"
          value="abcd@gmail.com"
        />
      </FormGroup>
    </Col>
    <Col
      md={6}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      <FormGroup
        tag="div"
      >
        <Label
          htmlFor="password"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
              "xxl",
            ]
          }
        >
          Password:
        </Label>
        <Input
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Enter the password"
          type="password"
          value="abcd"
        />
      </FormGroup>
    </Col>
  </Row>
</div>
`;
